nomenclate/core/nomenclature.py

```diff
@@ -2,146 +2,295 @@
-                                 e.g. - this_is_a_naming_string
-                                 the sections should be spaced around
+            format_target Union[str, list]: can be either a query path to a format
+                                            or in format of a naming string the sections should be spaced around
+                                              e.g. - this_is_a_naming_string
         Returns None: raises IOError if failure
         """
         try:
-            format_target = format_target if isinstance(format_target, list) else [format_target]
+            self._validate_format_string(format_target)
+            self.format_string = format_target
+        except exceptions.FormatError:
             try:
-                self._validate_format_string(format_target)
-                self.format_string = format_target
-            except:
+                format_target = format_target if isinstance(format_target, list) else [format_target]
+                self.format_string = self.cfg.get(format_target, return_type=str)
+            except exceptions.ResourceNotFoundError:
+                # TODO: Should log a default setting due to no target found.
                 self.format_string = self.cfg.get(self.DEFAULT_FORMAT_PATH, return_type=str)
-            self.format_order = self.get_format_order(self.format_string)
-            self.build_name_attrs()
-        #TODO: Custom error!!!!
-        except IOError:
-            raise IOError('Could not find naming format %s in config file nor was it a valid naming format' % format_target)
-
-    def initialize_options_from_config_file(self):
-        self.naming_formats = self.cfg.get('naming_formats', return_type=dict)
-        self.config_LUT = self.cfg.get('overall_config', return_type=dict)
-        self.suffix_LUT =  self.cfg.get('suffixes', return_type=dict)
+        finally:
+            self.format_order = self.get_format_order_from_format_string(self.format_string)
+            self.token_dict.build_name_attrs()
 
+    def initialize_options(self):
+        self.naming_formats = self.cfg.get(self.NAMING_FORMAT_PATH, return_type=dict)
+        self.config_table = self.cfg.get(self.CONFIG_PATH, return_type=dict)
+        self.suffix_table = self.cfg.get(self.SUFFIXES_FORMAT_PATH, return_type=dict)
 
     def initialize_ui_options(self):
         """
         Placeholder for all categories/sub-lists within options to be recorded here
         """
         pass
 
-    @staticmethod
-    def get_format_order(format_string):
+    def merge_dict(self, input_dict):
+        """ updates the current set of NameAttrs with new or overwritten values from a dictionary
+        Args:
+            kwargs (dict): any extra definitions you want to input into the resulting dictionary
+        Returns (dict): dictionary of relevant values
+        """
+        self.token_dict.state = input_dict
```

Comment:
auto magical...  Not a fan.  You can do this if you choose to, but this is extremely hard to read in comparison with something like `self.token_dict.update(input_dict)`